connector/connect/server/src/main/scala/org/apache/spark/sql/connect/artifact/SparkConnectArtifactManager.scala

```diff
@@ -161,7 +162,19 @@ class SparkConnectArtifactManager(sessionHolder: SessionHolder) extends Logging
    */
   def classloader: ClassLoader = {
     val urls = getSparkConnectAddedJars :+ classDir.toUri.toURL
-    new URLClassLoader(urls.toArray, Utils.getContextOrSparkClassLoader)
+    val loader = if (SparkEnv.get.conf.get(CONNECT_SCALA_UDF_STUB_CLASSES).nonEmpty) {
+      val stubClassLoader =
+        StubClassLoader(null, SparkEnv.get.conf.get(CONNECT_SCALA_UDF_STUB_CLASSES))
+      new ChildFirstURLClassLoader(
```

Comment:
Should this follow the same rules for classpath resolution we have on the executor?

Reply:
Probably should to be consistent. Let me fix in a followup.

Reply:
Actually it is fine. There are 3 existing class loader:

User CL : classes added using --jar
Sys CL: Spark + sys libs
Session CL: classes added using session.addArtifacts

In Executor:
* normal: Sys -> (User + Session) -> Stub
* reverse: (User + Session) -> Sys -> Stub 

In Driver:
* normal: (Sys + User) -> Session -> Stub
* reverse: (User -> Sys) -> Session -> Stub 

So here what you saw is () -> Session -> Stub.